vazhin/app.js

```diff
@@ -0,0 +1,283 @@
+cancelBtn.addEventListener("click", cancelTodo);
+
+// Functions
+
+function addTodo(event) {
+    event.preventDefault();
+
+    if (taskText.value === "") {
+
+        taskText.classList.add("fill-out-field");
+        taskText.placeholder = "add a task";
+        return;
+    }
+    else if (priorityInput.value === "") {
+
+        taskText.classList.remove("fill-out-field");
+        priorityInput.classList.add("fill-out-field");
+        priorityInput.placeholder = "select priority";
+        return;
+    }
+    else if (deadlineInput.value === "") {
+
+        priorityInput.classList.remove("fill-out-field");
+        deadlineInput.classList.add("fill-out-field");
+        deadlineInput.placeholder = "enter a deadline";
+        return;
+    }
+    else {
+        taskText.placeholder = "coding... 👩‍💻";
+        priorityInput.placeholder = '';
+        deadlineInput.placeholder = '';
+        deadlineInput.classList.remove("fill-out-field");
+
+        const objOfTask = {
+            task: taskText.value,
+            priority: priorityInput.value,
+            deadline: deadlineInput.value,
+        };
+
+        if (inEditMode) {
+            arrOfTasks.splice(indexOfInEditTask, 1, objOfTask);
+            inEditMode = false;
+            addBtn.innerHTML = "add task"
+        }
+        else {
+            arrOfTasks.push(objOfTask);
+        }
+    }
+
+    updateTasksInList();
+}
+
+function updateTasksInList() {
+
+    let previousTasks = document.querySelectorAll(".todo-list div");
+    for (let i = 0; i < previousTasks.length; i++) {
+        previousTasks[i].remove();
+    }
+
+    function compare(a, b) {
+        return a.priority - b.priority;
+    }
+
+    arrOfTasks.sort(compare);
+
+    for (let i = 0; i < arrOfTasks.length; i++) {
+
+        const todoDiv = document.createElement("div");
+        const task = document.createElement("li");
+        const priority = document.createElement("p");
+        const deadline = document.createElement("p");
+        const todoBtnsDiv = document.createElement("div");
+        const completedBtn = document.createElement('button');
+        const trashBtn = document.createElement('button');
+        const editBtn = document.createElement('button');
+
+        task.innerHTML = `task: ${arrOfTasks[i].task}`;
+        priority.innerHTML = `priority: ${arrOfTasks[i].priority}`;
+        deadline.innerHTML = `deadline: ${arrOfTasks[i].deadline}`;
+        completedBtn.innerHTML = '<i></i>';
+        trashBtn.innerHTML = '<i class="fas fa-trash-alt"></i>';
+        editBtn.innerHTML = 'edit';
+
+        noTasks.remove();
+        todoList.appendChild(todoDiv);
+        todoDiv.appendChild(task);
+        todoDiv.appendChild(priority);
+        todoDiv.appendChild(deadline);
+        todoDiv.appendChild(todoBtnsDiv);
+        todoBtnsDiv.appendChild(completedBtn);
+        todoBtnsDiv.appendChild(trashBtn);
+        todoBtnsDiv.appendChild(editBtn);
+
```

Comment:
while the code above and below this line can be used, it is a bit hard to debug and figure out how you are nesting the elements. it is better to do it using template HTML string, preferably another function that takes a todo object and returns HTML to be used like this:
`const task = document.createElement("li");`
`task.innerHTML = getHTML(arrOfTasks[I]);`

then your `getHTML()` function should return a string template like the following:

    function getHTML(todo) {
        return `<div class="todo-div">
            <p class="task-text">${todo.task}</p>
            <p class="display-5">${todo.priority}</p>
        </div>`; \\ you probably need to return more HTML but for simplicity I just id this now
    }